enhancements/insights/insights-config-api.md

```diff
@@ -0,0 +1,335 @@
+	// Messages is an optional attribute that provides error and warning messages
+	// from the gatherer
+	Messages []string `json:"messages,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// Reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,5,opt,name=reason"`
+}
+
+// GathererConditionType is a valid value for GathererCondition.Type
+type GathererConditionType string
+
+const (
+	// Gatherer was successful withnout any error reported
+	Successful GathererConditionType = "Successful"
+	// Gatherer was disabled by user
+	Disabled GathererConditionType = "Disabled"
+	// Gatherer failed to run
+	Failed GathererConditionType = "Failed"
+	// Gatherer was running, but there were some errors
+	Warning GathererConditionType = "Warning"
+)
+```
+
+### User Stories
+#### Force gathering - refresh Insights status
+As an OCP user/cluster-admin I would like to be able to trigger on demand gathering (there were some active Insights health checks in the cluster, which should be fixed and I want to refresh the Insights status). Steps:
+- the user will set a new value of `ForceGatherReason`
+- if all the gatherers are disabled (`disabledGatherers: ["all"]`), nothing happens
+- if there is some gathering in progress (`StartGatherTime` is not nil in the `GatheringStatus` section) then it will be interrupted and the new gathering will be triggered
+- if there is not any gathering in progress (`StartGatherTime` is nil in the `GatheringStatus` section) then the new gathering will be triggered almost immediately
+- user can check the `StartGatherTime` or the `LastGatheringTime` attribute in the `GatheringStatus` section to check if the new gathering already finished
+- when the gathering is finished then the value of `ForceGatherReason` is cleared and the `LastGatherReason` and `LastGatherTime` in the `GatheringStatus` are updated
+
+#### Disable Insights gathering completely
+As an OCP user/cluster-admin I would like to disable Insights gathering completely. Steps:
+- Set the `DisabledGatherers` to `["all"]` and the gathering will be turned off - i.e no new data will be gathered
+
+#### Exposing latest Insights status
+As an OCP user/cluster-admin I would like to see the latest Insights status. The status is represented by the `InsightsStatus` and it is only updated by the operator.
+
+#### Excluding particular gatherers
+As an OCP user/cluster-admin I would like to exclude particular gatherers. Steps:
+- Set the `DisabledGatherers` with the list of gatherers to be excluded/disabled - e.g `DisabledGatherers: ["clusterconfig/config_maps", "clusterconfig/node_logs"]` 
+
+### API Extensions
+
+This is adding new `insights.config.openshift.io` and `insights.operator.openshift.io` API resources. The proposal of the structure of the new API is mentioned in the [Proposal](#Proposal) section.
```

Comment:
To be precise the full name for the newly added resources should be:
- Insights in config.openshift.io/v1 API group
- Insights in operator.openshift.io/v1 API group 
:wink:

Reply:
Thanks. Updated.